### ISSUE TITLE
Calendar should announce the date in the order day, month, year (most specific information first).

### ISSUE THREAD

**Issue by JustSlone** [microsoft/fluentui#16072]
Calendar lists the month first in the aria-label for the days in the grid. This means that as a user navigates around the grid they hear the month over and over again. e.g. November 25th 2020, November 18th 2020, November 11th 2020. This requires the user to wait until the long month is announced before knowing what the date is. 

#### Actual behavior:

Currently Calendar has the aria-label attribute in this format: `aria-label="November 25, 2020"`. 

#### Expected behavior:

Calendar should have the aria-label attribute in this format: `aria-label="25 November 2020"`. day -> month -> year

### Priorities and help requested:

Are you willing to submit a PR to fix? No

Requested priority: Low

**Comment by jgbright** [comment]
I have a fix for this and would like to submit a PR, but I just noticed the pinned issue suggesting contributors hold off on submitting PRs until version 8 is finalized.  I'll just keep it locally for now.

**Comment by JustSlone** [comment]
@jgbright actually, the code is done moving around, so it's fine to submit a PR now. Go ahead and submit a PR against master, if needed the fix can be cherry-picked to 7.0 as well.

**Comment by khmakoto** [comment]
@JustSlone `Calendar` has the ability to overwrite the date formatter that is used when calculating the `aria-label` of the `CalendarDayGrid` as can be seen in [this codepen](https://codepen.io/khmakoto/pen/oNZNWrJ?editors=1010). So this is actually a non-issue.

**Comment by khmakoto** [comment]
@JustSlone and I talked offline and it seems the gist of this issue is more about changing the default more than anything else. The design for that seems to need more thought than initially perceived so we need further discussion with @lorejoh12 to know if there's a solution worth pursuing or not.

**Comment by khmakoto** [comment]
Had another offline chat with @JustSlone, @lorejoh12 and @smhigley about how to approach this. After we had several ideas floated we settled on composing the `aria-label` in `CalendarGridDayCell` ourselves in the following way `[day], [month], [year]` without making changes to the `dateTimeFormatter`. This allows us to prevent public-facing API changes while still solving the problem and paves the path for a more robust solution once `aria-description` becomes available.

**Comment by JustSlone** [comment]
Yeah that sounds about right @khmakoto. Another key point is that we decided that this is intentionally separated from localization. Irrespective of what the localized date format is, the aria-label will always be day, month, year. The aria-label should of course leverage the date strings to produce a string leveraging the appropriate localized names for things such as month.
